### PR TITLE
Simplify condition handling for DevWorkspaces

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Rework how conditions are synced for simplicity and avoid updating `LastTransitionTime` unless the condition's status actually has transitioned. 

### What issues does this PR fix or reference?
When testing DWO in `crc` with many DevWorkspaces running at once, I found that repeatedly updating the status caused every DevWorkspace to reconcile in a loop:

running-devworkspaces

(Effect is seen in the last 10 minutes of these metrics0

### Is it tested? How?
Test that conditions are set to `true|false|unknown` as expected.

### CODE REVIEW

File: controllers/workspace/status.go [hunk 1]
```diff
@@ -109,6 +81,51 @@ func (r *DevWorkspaceReconciler) updateWorkspaceStatus(workspace *dw.DevWorkspac
...
+			LastTransitionTime: currTransitionTime,
+			Type:               condType,
+			Status:             cond.Status,
+			Message:            cond.Message,
+		})
+	}
+
+	// Sort conditions to avoid unnecessary updates
+	sort.SliceStable(workspaceStatus.Conditions, func(i, j int) bool {
+		return strings.Compare(string(workspaceStatus.Conditions[i].Type), string(workspaceStatus.Conditions[j].Type)) > 0
```
Comment (sleshchenko) [inline controllers/workspace/status.go:125]: Would it be a good UX improvement if you order conditions in the order they are expected to appear?

**Review by sleshchenko — APPROVED** [review]

**Review by JPinkney — APPROVED** [review]

File: controllers/workspace/status.go (hunk 1 above)
Comment (amisevsk) [reply controllers/workspace/status.go]: :+1: Done

**Pull request merged**